misc/python/materialize/feature_benchmark/executor.py

```diff
@@ -46,3 +48,83 @@ def Kgen(self, topic: str, args: List[str]) -> Any:
         return self._composition.run(
             "kgen", f"--topic=testdrive-{topic}-{self._seed}", *args
         )
+
+
+class MzCloud(Executor):
+    def __init__(
+        self,
+        composition: Composition,
+        seed: int,
+        mzcloud_url: str,
+        kafka_addr: str,
+        schema_registry_url: str,
+    ) -> None:
+        self._composition = composition
+        self._seed = seed
+        self._mzcloud_url = mzcloud_url
+        self._kafka_addr = kafka_addr
+        self._schema_registry_url = schema_registry_url
+
+    def RestartMz(self) -> None:
+        # We can't restart the cloud, so complain.
+        assert False
+
+    def Reset(self) -> None:
+        print("resetting")
+        self.exec(
+            *[
+                "target/debug/testdrive",
+                f"--materialize-url={self._mzcloud_url}",
+                f"--seed={self._seed}",
+            ],
+            stdin="",
+        )
+
+    def Td(self, input: str) -> Any:
+        return self.exec(
+            *[
+                "target/debug/testdrive",
```

Comment:
debug testdrive is probably OK to use, but in the future it may suddenly get unacceptably slow for any reason (e.g. someone adds an expensive debug-only validation step in some piece of rust code) 

You may consider an alternative setup:

- run a Testdrive service in a container, as the default workflow
- call that Testdrive using the functionality provided by mzcompose, rather than implementing your own "exec" and construct a testdrive command-line argument list.

Reply:
Yes, I'd like this but didn't know how to do it, hence the hard coding of the executable.